src/detect-tls-subjectaltname.c

```diff
@@ -0,0 +1,208 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ * \file
+ *
+ * \author Shivani Bhardwaj <shivani@oisf.net>
+ *
+ * Implements support for tls.subjectaltname keyword.
+ */
+
+#include "suricata-common.h"
+#include "threads.h"
+#include "decode.h"
+#include "detect.h"
+
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-mpm.h"
+#include "detect-engine-prefilter.h"
+#include "detect-engine-content-inspection.h"
+#include "detect-content.h"
+#include "detect-tls-subjectaltname.h"
+#include "detect-engine-uint.h"
+
+#include "flow.h"
+#include "flow-util.h"
+#include "flow-var.h"
+
+#include "util-debug.h"
+#include "util-spm.h"
+#include "util-print.h"
+
+#include "stream-tcp.h"
+
+#include "app-layer.h"
+#include "app-layer-ssl.h"
+#include "util-profiling.h"
+
+static int DetectTlsSubjectAltNameSetup(DetectEngineCtx *, Signature *, const char *);
+static uint8_t DetectEngineInspectTlsSubjectAltName(DetectEngineCtx *de_ctx,
+        DetectEngineThreadCtx *det_ctx, const DetectEngineAppInspectionEngine *engine,
+        const Signature *s, Flow *f, uint8_t flags, void *alstate, void *txv, uint64_t tx_id);
+static int PrefilterMpmTlsSubjectAltNameRegister(DetectEngineCtx *de_ctx, SigGroupHead *sgh,
+        MpmCtx *mpm_ctx, const DetectBufferMpmRegistry *mpm_reg, int list_id);
+
+static int g_tls_subjectaltname_buffer_id = 0;
+
+typedef struct PrefilterMpmTlsSubjectAltName {
+    int list_id;
+    const MpmCtx *mpm_ctx;
+    const DetectEngineTransforms *transforms;
+} PrefilterMpmTlsSubjectAltName;
+
+/**
+ * \brief Registration function for keyword: tls.subjectaltname
+ */
+void DetectTlsSubjectAltNameRegister(void)
+{
+    sigmatch_table[DETECT_AL_TLS_SUBJECTALTNAME].name = "tls.subjectaltname";
+    sigmatch_table[DETECT_AL_TLS_SUBJECTALTNAME].desc =
+            "sticky buffer to match the TLS Subject Alternative Name buffer";
```

Comment:
Wondering: would it be crazy or too difficult to maybe have an automated way of picking this info to help populate the json schema, for keywords?

Reply:
We'll have to at least write one more script to direct the output from `list-keywords` output.
Or, there could be a logging fn in registration to take care of all logging related work. Not sure about this one if it's feasible.
But, nice thought. :) Out of scope of this PR though. But, maybe good to pitch to others and take their opinion.